x-pack/plugins/security_solution/public/resolver/view/panels/panel_content_related_detail.tsx

```diff
@@ -0,0 +1,340 @@
+    max-width: 8em;
+  }
+  &.euiDescriptionList.euiDescriptionList--column dd.euiDescriptionList__description {
+    max-width: calc(100% - 8.5em);
+    overflow-wrap: break-word;
+  }
+`);
+
+// Styling subtitles, per UX review:
+const StyledFlexTitle = memo(styled('h3')`
+  display: flex;
+  flex-flow: row;
+  font-size: 1.2em;
+`);
+const StyledTitleRule = memo(styled('hr')`
+  &.euiHorizontalRule.euiHorizontalRule--full.euiHorizontalRule--marginSmall.override {
+    display: block;
+    flex: 1;
+    margin-left: 0.5em;
+  }
+`);
+
+const TitleHr = memo(() => {
+  return (
+    <StyledTitleRule className="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginSmall override" />
+  );
+});
+TitleHr.displayName = 'TitleHR';
+
+/**
+ * This view presents a detailed view of all the available data for a related event, split and titled by the "section"
+ * it appears in the underlying ResolverEvent
+ */
+export const RelatedEventDetail = memo(function RelatedEventDetail({
+  relatedEventId,
+  parentEvent,
+  pushToQueryParams,
+  countForParent,
+}: {
+  relatedEventId: string;
+  parentEvent: ResolverEvent;
+  pushToQueryParams: (arg0: CrumbInfo) => unknown;
+  countForParent: number | undefined;
+}) {
+  const processName = (parentEvent && event.eventName(parentEvent)) || '*';
+  const processEntityId = parentEvent && event.entityId(parentEvent);
+  const totalCount = countForParent || 0;
+  const eventsString = i18n.translate(
+    'xpack.securitySolution.enpoint.resolver.panel.relatedEventDetail.events',
+    {
+      defaultMessage: 'Events',
+    }
+  );
+  const naString = i18n.translate(
+    'xpack.securitySolution.enpoint.resolver.panel.relatedEventDetail.NA',
+    {
+      defaultMessage: 'N/A',
+    }
+  );
+
+  const relatedsReadyMap = useSelector(selectors.relatedEventsReady);
+  const relatedsReady = relatedsReadyMap.get(processEntityId!);
+  const dispatch = useResolverDispatch();
+
+  /**
+   * If we don't have the related events for the parent yet, use this effect
+   * to request them.
+   */
+  useEffect(() => {
+    if (typeof relatedsReady === 'undefined') {
+      dispatch({
+        type: 'userRequestedRelatedEventData',
+        payload: processEntityId,
+      });
+    }
+  }, [relatedsReady, dispatch, processEntityId]);
+
+  const relatedEventsForThisProcess = useSelector(selectors.relatedEventsByEntityId).get(
+    processEntityId!
+  );
+
+  const [relatedEventToShowDetailsFor, countBySameCategory, relatedEventCategory] = useMemo(() => {
+    if (!relatedEventsForThisProcess) {
+      return [undefined, 0];
+    }
+    const specificEvent = relatedEventsForThisProcess.events.find(
+      (evt) => event.eventId(evt) === relatedEventId
+    );
+    // For breadcrumbs:
+    const specificCategory = specificEvent && event.eventType(specificEvent);
+    const countOfCategory = relatedEventsForThisProcess.events.reduce((sumtotal, evt) => {

```

Comment:
Would the stats from the backend work here?

Reply:
It's using the stats from the backend, but the ones `byCategory`.Thinking about it - that's probably not accurate since [file, dns] would count twice. Good catch, I'll change it.

Reply:
But yeah, now that I think about it *further* I remember that we thought it would be confusing to display
> 3 total
>2 dns
>2 file
> 1 registry
for [dns,file], [dns,file], [registry]

Reply:
The way it lies here reflects that the above case would be handled by showing:
>5 total
>2 dns
>2 file
>1 registry
So you would click on `file` and get the 2 [dns,file] [dns,file] entries displayed (although to your earlier point, they will display with descriptive names for `file` in that case. I didn't want to load too much functionality into the `descriptiveName` stuff without reviewing it with the team first... one idea, I guess, would be to extend the `descriptiveName` with a hint argument for _which_ descriptive name you want. So `event.descriptiveName(event, 'dns')` would give you the dns pretty name and `event.descriptiveName(event, `file`)` would give you the file name but like I said, I want to go over that with the team first and I would probably need some help from @oatkiller because I've maxed out my knowledge of TS as it is on that.